packages/nimble-components/src/rich-text-editor-viewer/specs/README.md

```diff
@@ -0,0 +1,373 @@
+Therefore, it is essential to develop web components that allow users to view and create rich text content, enabling their use in various scenarios,
+including Comments and other instances that necessitate rich text capabilities.
+
+[Nimble issue #1288](https://github.com/ni/nimble/issues/1288)
+
+[Comments UI mockup](https://www.figma.com/file/Q5SU1OwrnD08keon3zObRX/SystemLink?type=design&node-id=6280-94045)
+
+### Non-goals
+
+-   The visual design and interaction design for these components are out of scope for the initial release. However we have an
+    [IxD workflow](https://www.figma.com/file/Q5SU1OwrnD08keon3zObRX/SystemLink?type=design&node-id=6280-94045) for SLE application that we will rely
+    on.
+-   Blazor integration will be out of scope for this component as we have less or no experience in Blazor component development or other
+    related technology. Blazor example app addition will also be not covered in this spec.
+-   Adding an example Angular client application for this component may not be included in the initial releases. However, the integration
+    of the rich-text-editor with Angular applications, achieved through a directive, will be developed following the completion of this
+    component.
+
+### Features
+
+The `nimble-rich-text-editor` supports the basic text formatting options that includes:
+
+1. Bold
+2. Italics
+3. Numbered List
+4. Bulleted List
+
+This component will also offer APIs and interactive methods to format texts in the following ways:
+
+1. Using the buttons located below the editor to modify the text formatting for the selected texts.
+2. Using the keyboard shortcuts to modify the text formatting in the editor.
+3. Support to get markdown output from the editor.
+4. Support to convert the markdown to HTML to render it back in the `nimble-rich-text-viewer` component.
+
+#### _Additional features out of scope of this spec_
+
+-   Allowing the user to tag or mention by entering `@` in the editor and selecting the user name from the drop-down list.
+-   Support for adding images to the editor either by uploading or by pasting it.
+-   Support for adding hyperlinks to the existing words in the editor. However, adding raw links by pasting to the editor is supported.
+-   Support for striking out and underlining text. The reason for including these features in future scope that we utilize the
+    [prosemirror-markdown](https://github.com/ProseMirror/prosemirror-markdown) serializer and parser to convert the text into markdown format and vice
+    versa. However, the supported functionality of prosemirror-markdown, as mentioned in their
+    [documentation](https://github.com/ProseMirror/prosemirror-markdown#documentation), does not include support for strikeout and underline. To
+    address this limitation, we need to extend the class and add the necessary nodes to both the serializer and parser which will be added in the
```

Comment:
Is there any tiptap/prose mirror documentation that instructs on applying these extensions for strikeout and underline behaviors? If yes, it will be good to link it here.